### DISCUSSION TITLE
Klipper-Adaptive-Meshing-Purging (KAMP) incorporated

### DISCUSSION THREAD

**Discussion by bassamanator** [bassamanator/Sovol-SV06-firmware#75] [Announcements]
# Printers

- Sovol SV06
- ~~Sovol SV06 Skr-Mini-E3-V3.0~~
- ~~Sovol SV06 Plus~~

Will push to all other branches after testing in `develop` is successful.

# Changelog

- Imports in printer.cfg adjusted.
- KAMP can be enabled/disabled from within ` [gcode_macro _globals]`; enabled by default only in this release/branch.
- Other minor code cleanup.

# Installation

- Standard instructions in `master` `README.md` can be followed, however, relevant code lives in the `develop` branch, be sure to pull the correct branch.

<hr /><em>This discussion was created from the release <a href='https://github.com/bassamanator/Sovol-SV06-firmware/releases/tag/v0.1.3-beta1'>Klipper-Adaptive-Meshing-Purging (KAMP) incorporated</a>.</em>

**Comment by Manfredo1952** [comment]
Good day - how's the progress re testing this with the SV06 and possible transfer to SV06+,   please ?

**Reply by bassamanator** [reply]
KAMP is working fine on my end, no reports from other users. It's really not an issue, you can easily disable it via the `_globals` marcro/section. If you're comfortable with these things, feel free to pull all code from `develop` branch, except for the `printer.cfg` which you should use from either the `sv06-plus` branch, or use your existing `printer.cfg`.

In any case, I'll be merging KAMP into all branches at some point this week.

**Comment by Carpenter64** [comment]
Hi,

I use it on SV06+ since I saw you were including it in the SV06 branch a
few days ago.
It's working great. I really like it ! Great idea to include it in all
branches.

Have a nice day !

**Reply by bassamanator** [reply]
Thanks for the feedback!

**Comment by banhathome** [comment]
Do we still need to run `DO_CREATE_MESH` if we're using KAMP?

**Reply by bassamanator** [reply]
Well the instructions are for the initial setup, so no, this step **cannot** be skipped. Moreover, if kamp is disabled, and you don't have a `default` mesh, `print_start` will crash. There are instructions on how to disable loading a default mesh, search for `NOTE` in `print_start`.

I pushed KAMP to `master` last night, instructions on how to enable still not included in the readme. Here's how to enable (KAMP is disabled by default).
```
##### How do I enable KAMP (Klipper Adaptive Meshing & Purging)?

Although this repo contains all the code from the KAMP repository, only the `mesh` functionality of KAMP has been enabled and tested.

Adjust according to your needs.

| File     | `cfgs/misc-macros.cfg`   |
| -------- | ------------------------ |
| Section  | `[gcode_macro _globals]` |
| Variable | `variable_kamp_enable`   |
| Disable  | `0`                      |
| Enable   | `1`                      |
```

**Reply by bassamanator** [reply]
Also wanted to mention, if kamp is enabled, the default mesh isn't loaded.
